library/gcm.c

```diff
@@ -386,22 +430,32 @@ int mbedtls_gcm_update( mbedtls_gcm_context *ctx,
             return( ret );
         }
 
+        for(i=0; i<16; i++){
+        	ctx->ectr_remain[i]=ectr[i];
+        }
+
         for( i = 0; i < use_len; i++ )
         {
             if( ctx->mode == MBEDTLS_GCM_DECRYPT )
                 ctx->buf[i] ^= p[i];
-            out_p[i] = ectr[i] ^ p[i];
+            out_p[i + pre_len] = ectr[i] ^ p[i];
             if( ctx->mode == MBEDTLS_GCM_ENCRYPT )
-                ctx->buf[i] ^= out_p[i];
+                ctx->buf[i] ^= out_p[i+pre_len];
         }
-
-        gcm_mult( ctx, ctx->buf, ctx->buf );
+        ctx->data_remain = 16 - use_len;
+        
+		if( ctx->data_remain == 0 )
```

Comment:
fix indentation. Did you use tabs here?

Reply:
Please replace all uses of tabs by whitespaces.